library/PreseedControl.py

```diff
@@ -0,0 +1,117 @@
+        fetch the thing via the regular call in the Ansible role"""
+
+    def __init__(self, requester, preseed_name):
+        self.requester = requester
+        self.name = preseed_name
+
+    def _get_preseed_from_file(self, preseed_file, preseed_desc, preseed_type, public=False,
+                               knowngood=False):
+        json_preseed = {}
+        contents = ''
+
+        with open(preseed_file, 'r') as fd:
+            lines = fd.readlines()
+
+        for line in lines:
+            contents += line
+
+        json_preseed['content'] = contents
+        json_preseed['name'] = self.name
+        json_preseed['type'] = preseed_type
+        json_preseed['public'] = public == 'True'
+        json_preseed['known_good'] = knowngood == 'True'
+        if preseed_desc != '':
+            json_preseed['description'] = preseed_desc
+
+        return json_preseed
+
+    def check_for_existence(self):
+        url = '/api/v1/preseed?show_all=true'
+
+        try:
+            preseeds = self.requester.get(url)
+        except Exception as err:
+            raise err
+
+        for preseed in preseeds:
+            if preseed['name'] == self.name:
+                return preseed
+
+        return False
+
+    def get_preseed_id(self):
+        try:
+            preseed = self.check_for_existence()
+        except Exception as err:
+            raise err
+
+        if preseed != False:
+            return preseed['id']
+        else:
+            return None
+
+    def upload_preseed(self, preseed_file, preseed_type, preseed_desc=None,
+                       public=False, knowngood=False):
+        """Uploads preseed. Should check first that preseed doesn't exist, else
+        it modifies preseed (separate function ?). Post to upload, Put to
+        modify (+ id). Maybe implement a jinja2 syntax check ? But that should
+        be done on mrp's side"""
+
+        preseed_id = None
+        preseed_id = self.get_preseed_id()
+
+        if preseed_id is None and preseed_file == '':
+            raise ProvisionerError('Preseed does not exist and file not given')
+
+        if preseed_id is not None and preseed_file != '':  # Exists and file given
+            try:
+                return self._modify_preseed(preseed_id, 'PUT', preseed_file,
+                                           preseed_type, preseed_desc, public,
+                                           knowngood)
+            except ProvisionerError as err:
+                raise err
+
+        elif preseed_id is None and preseed_file != '':  # Doesn't exist and file given
+            try:
+                return self._modify_preseed(preseed_id, 'POST', preseed_file,
+                                           preseed_type, preseed_desc, public,
+                                          knowngood)
+            except ProvisionerError as err:
+                raise err
+
+        else:  # Exists and file not given, is it useful fetching contents?
```

Comment:
This will encourage bad usage (check for return == "some string"). If file is not given, just return.

If we need to check for existence, we can already use the get_preseed_id method.